tests/ci/integration/openvpn_patch/aws-lc-openvpn2-6-x.patch

```diff
@@ -38,10 +38,22 @@ index c9fa7196..a48ef391 100644
  #endif
  
 diff --git a/src/openvpn/ssl_openssl.c b/src/openvpn/ssl_openssl.c
-index 2595f878..cf99c3ec 100644
+index 23e76231..cab58ff4 100644
 --- a/src/openvpn/ssl_openssl.c
 +++ b/src/openvpn/ssl_openssl.c
-@@ -1658,8 +1658,10 @@ tls_ctx_load_ca(struct tls_root_ctx *ctx, const char *ca_file,
+@@ -1431,7 +1431,11 @@ tls_ctx_use_external_ec_key(struct tls_root_ctx *ctx, EVP_PKEY *pkey)
+ 
+     /* Among init methods, we only need the finish method */
+     EC_KEY_METHOD_set_init(ec_method, NULL, openvpn_extkey_ec_finish, NULL, NULL, NULL, NULL);
++#ifdef OPENSSL_IS_AWSLC
++    EC_KEY_METHOD_set_sign(ec_method, ecdsa_sign, NULL, ecdsa_sign_sig);
++#else
+     EC_KEY_METHOD_set_sign(ec_method, ecdsa_sign, ecdsa_sign_setup, ecdsa_sign_sig);
++#endif
```

Comment:
Instead of using the `OPENSSL_IS_AWSLC` macro, could we have it always pass NULL for that parameter?  Would that cause a failure when building against OpenSSL?
```
EC_KEY_METHOD_set_sign(ec_method, ecdsa_sign, NULL, ecdsa_sign_sig);
```

Reply:
Eventually when we upstream, the goal is to get OpenVPN code to have the "LC variant" with NULL passed in. This shouldn't cause an error in OpenSSL either. It's written this way just for the patch